account_invoice_change_currency/hooks.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from odoo import fields, tools
+
+_logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(cr):
+    """
+    Initializing column custom_rate on table account_move
+    for the improvement in performance to avoid long
+    duration in databases with thousand of moves
+    """
+    _logger.info("Initializing column custom_rate on table account_move")
+    tools.create_column(
+        cr=cr,
+        tablename="account_move",
+        columnname="custom_rate",
+        columntype=fields.Float().column_type[1],
```

Comment:
Well, without parenthesis gives you "property object is not subscriptable"

Reply:
I see. That's because float fields [don't use always the same value](https://github.com/odoo/odoo/blob/a66bed194cfd1adf8dd4030448fc5ccd9dc5d3c1/odoo/fields.py#L1313), because it depends on whether digits is defined.

Since we know in advance the field has digits, let's use `numeric`

Reply:
@luisg123v Done!